src/client/components/pages/Courses/OfferedModal.tsx

```diff
@@ -0,0 +1,228 @@
+    offered: '',
+  });
+
+  const updateFormFields = (event: ChangeEvent): void => {
+    const target = event.target as FormField;
+    setFormFields({
+      ...form,
+      [target.name]: target.value,
+    });
+  };
+
+  const instanceIdentifier = currentCourseInstance && currentSemester
+    ? `${
+      currentCourseInstance.catalogNumber
+    }, ${
+      currentSemester.term
+    } ${
+      currentSemester.calendarYear
+    }`
+    : '';
+
+  useEffect(() => {
+    if (currentSemester && currentCourseInstance) {
+      const currentTermKey = currentSemester.term.toLowerCase() as TermKey;
+      setFormFields({
+        offered: currentCourseInstance[currentTermKey].offered,
+      });
+    }
+  }, [currentSemester, currentCourseInstance]);
+
+  useEffect(() => {
+    if (isVisible) {
+      setOfferedModalFocus();
+      setErrorMessage('');
+    }
+  }, [isVisible]);
+
+  const saveOffered = async () => {
+    setSaving(true);
+    let results: CourseInstanceUpdateDTO;
+    try {
+      const currentTermKey = currentSemester.term.toLowerCase() as TermKey;
+      results = await updateCourseInstance(
+        currentCourseInstance[currentTermKey].id,
+        {
+          offered: form.offered as OFFERED,
+          preEnrollment: currentCourseInstance[currentTermKey].preEnrollment,
+          studyCardEnrollment: currentCourseInstance[currentTermKey]
+            .studyCardEnrollment,
+          actualEnrollment: currentCourseInstance[currentTermKey]
+            .actualEnrollment,
+        }
+      );
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        const serverError = error.response.data as Error;
+        setErrorMessage(serverError.message);
+      } else if (results === undefined) {
+        const { message } = error as Error;
+        setErrorMessage(`Failed to save meeting data. Please try again later.
+        ${message}`);
+      }
+    } finally {
+      setSaving(false);
+    }
+    if (results) {
+      onSave(results);
+    }
+  };
+
+  return (
+    <Modal
+      ariaLabelledBy="edit-offered-modal"
+      closeHandler={onClose}
+      isVisible={isVisible}
+    >
+      <ModalHeader
+        forwardRef={modalHeaderRef}
+        tabIndex={0}
+      >
+        <span id="edit-offered-header">
+          {`Edit Offered Value for ${instanceIdentifier}`}
+        </span>
+      </ModalHeader>
+      <ModalBody>
+        {saving
+          ? (
+            <LoadSpinner>Saving</LoadSpinner>
+          )
+          : (
+            <Form
+              id="editOfferedForm"
+              label="Edit Offered Form"
+            >
+              <Dropdown
+                id="offered"
+                name="offered"
+                label="Edit Offered Value Dropdown"
+                options={
+                  Object.values(OFFERED)
+                    .filter((value) => value !== OFFERED.RETIRED)
+                    .map((offeredValue): {
+                      value: string; label: string
```

Comment:
Technically we don't need the return type here because it can be inferred by the immediate return value, but since we have it we should have `value: OFFERED`.